x-pack/plugins/observability_solution/slo/public/embeddable/slo/overview/slo_configuration.tsx

```diff
@@ -7,81 +7,106 @@
+  const hasGroupBy = selectedSlo && selectedSlo.sloInstanceId !== ALL_VALUE;
 
   const onConfirmClick = () =>
     onCreate({
       showAllGroupByInstances,
       sloId: selectedSlo?.sloId,
       sloInstanceId: selectedSlo?.sloInstanceId,
+      overviewMode,
     });
-  const [hasError, setHasError] = useState(false);
 
   return (
-    <EuiModal onClose={onCancel} style={{ minWidth: 550 }}>
-      <EuiModalHeader>
-        <EuiModalHeaderTitle>
-          {i18n.translate('xpack.slo.sloEmbeddable.config.sloSelector.headerTitle', {
-            defaultMessage: 'SLO configuration',
-          })}
-        </EuiModalHeaderTitle>
-      </EuiModalHeader>
-      <EuiModalBody>
+    <>
+      <EuiFlyoutBody>
         <EuiFlexGroup>
-          <EuiFlexItem grow>
-            <SloSelector
-              singleSelection={true}
-              hasError={hasError}
-              onSelected={(slo) => {
-                setHasError(slo === undefined);
-                if (slo && 'id' in slo) {
-                  setSelectedSlo({ sloId: slo.id, sloInstanceId: slo.instanceId });
-                }
-              }}
-            />
+          <EuiFlexItem>
+            <EuiFlexGroup>
+              <EuiFlexItem>
+                <SloSelector
+                  singleSelection={true}
+                  hasError={hasError}
+                  onSelected={(slo) => {
+                    setHasError(slo === undefined);
+                    if (slo && 'id' in slo) {
+                      setSelectedSlo({ sloId: slo.id, sloInstanceId: slo.instanceId });
+                    }
+                  }}
+                />
+              </EuiFlexItem>
+            </EuiFlexGroup>
+            {hasGroupBy && (
+              <>
+                <EuiSpacer />
+                <EuiSwitch
+                  label={i18n.translate(
+                    'xpack.slo.sloConfiguration.euiSwitch.showAllGroupByLabel',
+                    {
+                      defaultMessage: 'Show all related group-by instances',
+                    }
+                  )}
+                  checked={showAllGroupByInstances}
+                  onChange={(e) => {
+                    setShowAllGroupByInstances(e.target.checked);
+                  }}
+                />
+              </>
+            )}
           </EuiFlexItem>
         </EuiFlexGroup>
-        {selectedSlo?.sloInstanceId !== ALL_VALUE && (
-          <>
-            <EuiSpacer />
-            <EuiSwitch
-              label={i18n.translate('xpack.slo.sloConfiguration.euiSwitch.showAllGroupByLabel', {
-                defaultMessage: 'Show all related group-by instances',
-              })}
-              checked={showAllGroupByInstances}
-              onChange={(e) => {
-                setShowAllGroupByInstances(e.target.checked);
-              }}
-            />
-          </>
-        )}
-      </EuiModalBody>
-      <EuiModalFooter>
+      </EuiFlyoutBody>
+      <EuiFlyoutFooter>
```

Comment:
Minor suggestion for spacing out the cancel and save buttons:

```suggestion
      <EuiFlyoutFooter>
        <EuiFlexGroup justifyContent="spaceBetween">
```